src/fieldPresets/checkbox.js

```diff
@@ -1,10 +1,10 @@
 export default {
   valuePropName: 'checked',
   initialValue: false,
-  mapPropsToField: ({ props: { checked }, fieldRecord }) => ({
+  mapPropsToField: ({ props: { checked, initialValue }, fieldRecord }) => ({
     ...fieldRecord,
     type: 'checkbox',
-    checked: !!checked,
+    checked: !!initialValue || !!checked,
```

Comment:
In the original issue you've experienced, was it the initial value of `checked` that was wrong?

I somehow understood that the issue was that checkbox is always treated as uncontrolled:

https://github.com/kettanaito/react-advanced-form/blob/c276e6c715c5d68602aa54088f7adc210b599663/src/components/createField.jsx#L142-L144

I'm verifying because there appears to be no `initialValue` for `input[type="checkbox"]` in React. It's correct to use `defaultChecked` instead for uncontrolled checkboxes.

What would you say if we changed `initialValue` to `defaultChecked` here? This would be the same in terms of propagating proper initial value of an uncontrolled checkbox to form's state. Then we could change the value of the `controlled` property in `createField.js` linked above to something like this:

```js
// createField.js
controlled: prunedProps.hasOwnProperty(valuePropName)
```

The only thing that would still fall off with this, is that `getInitialValue` function in `createField` always takes `fieldProps.initialValue`. For checkbox scenario that would be `defaultChecked` then. I'm not sure what to favor here: the default flow one would use in React (even if it means diverging in initial value checks), or use unified `initialValue` provided by the library's API.

Please, what do you think about this?

Reply:
I understand what you mean by that. Maybe stick with initialValue will be good enough. Simple API is a key :)
And the problem was that i was setting initialValues for the Form itself and checkbox.js wasn't counting with that value. Form component provided properly initialValue to prunedProps but fieldPreset/checkbox wasn't counting with that. Maybe other form components can have the same problem. That will be fine to make a tests for it.

Reply:
Agree, let's keep the same API throughout different field types.

May I ask you to check why the [current integration tests for `initialValues`](https://github.com/kettanaito/react-advanced-form/blob/master/cypress/integration/basics/InitialValues.spec.js) didn't catch your use case? I'd like to add the missing test(s) as a part of this pull request. Thank you.